sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncBatch.java

```diff
@@ -0,0 +1,385 @@
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.data.tables.implementation.BatchImpl;
+import com.azure.data.tables.implementation.ModelHelper;
+import com.azure.data.tables.implementation.TablesMultipartSerializer;
+import com.azure.data.tables.implementation.models.BatchChangeSet;
+import com.azure.data.tables.implementation.models.BatchOperation;
+import com.azure.data.tables.models.BatchOperationResponse;
+import com.azure.data.tables.implementation.models.BatchRequestBody;
+import com.azure.data.tables.implementation.models.BatchSubRequest;
+import com.azure.data.tables.implementation.models.TableServiceError;
+import com.azure.data.tables.implementation.models.TableServiceErrorException;
+import com.azure.data.tables.models.TableEntity;
+import com.azure.data.tables.models.UpdateMode;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+/**
+ * Provides a batch object for asynchronously executing a transaction containing one or more operations on entities
+ * within a table in the Azure Tables service.
+ *
+ * The batch object represents a collection of one or more create, update, upsert, and/or delete operations on entities
+ * that share the same partition key within the table. When the batch is executed, all of the operations will be
+ * performed as part of a single transaction. As a result, either all operations in the batch will succeed, or if a
+ * failure occurs, all operations in the batch will be rolled back. Each operation in a batch must operate on a distinct
+ * row key. Attempting to add multiple operations to a batch that share the same row key will cause an exception to be
+ * thrown.
+ *
+ * Instances of this object are obtained by calling the {@link TableAsyncClient#createBatch(String)} method on a {@link
+ * TableAsyncClient} object.
+ */
+@Fluent
+public final class TableAsyncBatch {
+    private final ClientLogger logger = new ClientLogger(TableAsyncBatch.class);
+    private final String partitionKey;
+    private final TableAsyncClient operationClient;
+    private final BatchImpl batchImpl;
+    private final HashSet<String> rowKeys = new HashSet<>();
+    private final List<BatchOperation> operations = new ArrayList<>();
+    private boolean frozen = false;
+
+    TableAsyncBatch(String partitionKey, TableAsyncClient client) {
+        this.partitionKey = partitionKey;
+        this.batchImpl = new BatchImpl(client.getImplementation(), new TablesMultipartSerializer());
```

Comment:
Do we need a new instance of this serializer for each batch? Can it be passed in as a dependency?

Reply:
Hmm, you're probably right. I think I was following the pattern from our `TableClientBuilder` - each builder instance has its own instance of `TablesJacksonSerializer` and configures the client with that serializer when `build()` is called. Would a lazy static be a better / more efficient way to handle this?